packages/core/src/messagingProtocols/mls/E2EIdentityService/E2EIServiceExternal.ts

```diff
@@ -47,6 +47,24 @@ type Events = {
   crlChanged: {domain: string};
 };
 
+// TODO coreCrypto types are wrong here. They return a string (the key of the enum) instead of the enum value
+function fixDeviceStatus(value: any): DeviceStatus {
+  const fixedValue = DeviceStatus[value];
+  if (!fixedValue) {
+    throw new Error(`Invalid device status: ${value}`);
+  }
+  return fixedValue as unknown as DeviceStatus;
+}
+
+// TODO coreCrypto types are wrong here. They return a string (the key of the enum) instead of the enum value
+function fixConversationState(value: any): E2eiConversationState {
+  const fixedValue = E2eiConversationState[value];
+  if (!fixedValue) {
+    throw new Error(`Invalid conversation status: ${value}`);
+  }
+  return fixedValue as unknown as E2eiConversationState;
+}
```

Comment:
Those are temporary functions that can be removed when CC fixes the type mismatch